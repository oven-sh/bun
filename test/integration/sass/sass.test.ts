import { compileString } from "sass";

test("sass source maps", () => {
  const scssString = `.ruleGroup {
    display: flex;
    flex-direction: column;
    gap: 0.5rem;
    padding: 0.5rem;
    border-width: 1px;
  }
  `;

  expect(compileString(scssString, { sourceMap: false })).toMatchSnapshot();
  expect(compileString(scssString, { sourceMap: true })).toMatchSnapshot();
});
