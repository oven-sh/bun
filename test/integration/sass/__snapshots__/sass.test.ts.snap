// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`sass source maps 1`] = `
{
  "css": 
".ruleGroup {
  display: flex;
  flex-direction: column;
  gap: 0.5rem;
  padding: 0.5rem;
  border-width: 1px;
}"
,
  "loadedUrls": [],
}
`;

exports[`sass source maps 2`] = `
{
  "css": 
".ruleGroup {
  display: flex;
  flex-direction: column;
  gap: 0.5rem;
  padding: 0.5rem;
  border-width: 1px;
}"
,
  "loadedUrls": [],
  "sourceMap": {
    "mappings": "AAAA;EACI;EACA;EACA;EACA;EACA",
    "names": [],
    "sourceRoot": "",
    "sources": [
      "data:;charset=utf-8,.ruleGroup%20%7B%0A%20%20%20%20display:%20flex;%0A%20%20%20%20flex-direction:%20column;%0A%20%20%20%20gap:%200.5rem;%0A%20%20%20%20padding:%200.5rem;%0A%20%20%20%20border-width:%201px;%0A%20%20%7D%0A%20%20",
    ],
    "version": 3,
  },
}
`;
